runtime/ms_rest_azure/lib/ms_rest_azure/credentials/msi_token_provider.rb

```diff
@@ -93,13 +119,29 @@ def acquire_token
       end
 
       fail AzureOperationError,
-        'Couldn\'t acquire access token from Managed Service Identity, please verify your tenant id, port and settings' unless response.status == 200
+          'Couldn\'t acquire access token from Managed Service Identity, please verify your tenant id, port and settings' unless response.status == 200
 
       response_body = JSON.load(response.body)
       @token = response_body['access_token']
       @token_expires_on = Time.at(Integer(response_body['expires_on']))
       @token_type = response_body['token_type']
     end
+
+    #
+    # Sets user assigned identity value in request body
+    # @param request_body [String] body of the request used to acquire token
+    # @param id_type [String] type of id to send 'client_id', 'object_id' or 'msi_res_id'
+    # @param id_value [String] id of the user assigned identity
+    #
+    # @return [String] new authentication token.
```

Comment:
In case it helps, the comment can be improved be `new request body with with the addition of <id_type>=<id_value >`?